### PR TITLE
shallow import for `gluonts.mx` module

### PR DESCRIPTION
**Pull request by Schmedu**
*Issue #, if available:*
https://github.com/awslabs/gluon-ts/issues/1196
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
@Schmedu are these the only import paths that can be reasonably shortened? One way to check this, is to go through the tutorial notebooks (which are generated from the markdown files [here](https://github.com/awslabs/gluon-ts/tree/master/docs/tutorials) and see what imports are being used. We should also shorten the imports in the tutorials themselves.

**Comment by lostella** [comment]
For example in the extended forecasting tutorial there is 

```python
from gluonts.mx.batchify import as_in_context, batchify
from gluonts.mx.model.estimator import GluonEstimator
from gluonts.mx.model.predictor import RepresentableBlockPredictor
from gluonts.mx.trainer import Trainer
from gluonts.mx.util import copy_parameters, get_hybrid_forward_input_names
```

so maybe more stuff can be exposed

**Comment by Schmedu** [comment]
> @Schmedu are these the only import paths that can be reasonably shortened? One way to check this, is to go through the tutorial notebooks (which are generated from the markdown files [here](https://github.com/awslabs/gluon-ts/tree/master/docs/tutorials) and see what imports are being used. 

I did not update my last analyses about the import paths, but will do so now.

> We should also shorten the imports in the tutorials themselves.

That's also a very good point, thanks!

File: docs/tutorials/forecasting/extended_tutorial.md [hunk 1]
```diff
@@ -845,11 +845,11 @@ from mxnet.gluon import HybridBlock
...
-from gluonts.mx.batchify import as_in_context, batchify
-from gluonts.mx.model.estimator import GluonEstimator
-from gluonts.mx.model.predictor import RepresentableBlockPredictor
-from gluonts.mx.trainer import Trainer
-from gluonts.mx.util import copy_parameters, get_hybrid_forward_input_names
+from gluonts.mx import as_in_context, batchify
+from gluonts.mx import GluonEstimator
+from gluonts.mx import RepresentableBlockPredictor
+from gluonts.mx import Trainer
+from gluonts.mx import copy_parameters, get_hybrid_forward_input_names
```
Comment (lostella) [inline docs/tutorials/forecasting/extended_tutorial.md:852]: I guess these could be all in one? As in

```
from gluonts.mx import (
   ...
   ...
)
```

File: src/gluonts/mx/__init__.py [hunk 2]
```diff
@@ -24,3 +24,27 @@
...
 
 from . import prelude as _  # noqa
+from .trainer import Trainer  # noqa
+from .util import get_hybrid_forward_input_names, copy_parameters  # noqa
+from .kernels import RBFKernel  # noqa
+from .batchify import as_in_context, batchify  # noqa
+from .model.estimator import GluonEstimator  # noqa
+from .model.predictor import RepresentableBlockPredictor  # noqa
+from .distribution import DistributionOutput, GaussianOutput  # noqa
+from .block.scaler import MeanScaler, NOPScaler  # noqa
```
Comment (lostella) [inline src/gluonts/mx/__init__.py:34]: We could sort the imports here (`isort` to the rescue!)

File: src/gluonts/mx/__init__.py [hunk 3]
```diff
@@ -24,3 +24,27 @@
...
+    "GaussianOutput",
+    "RepresentableBlockPredictor",
+    "GluonEstimator",
+    "as_in_context",
+    "batchify",
+    "copy_parameters",
+    "RBFKernel",
+    "get_hybrid_forward_input_names",
+    "Trainer",
+]
```
Comment (lostella) [inline src/gluonts/mx/__init__.py:50]: I think this is needed only if we want to *exclude* stuff from `__all__`. Is that the case? For example, this way `Tensor` (defined above) is not visible in `gluonts.mx`

**Review by lostella — CHANGES_REQUESTED** [review]

File: src/gluonts/mx/__init__.py (hunk 3 above)
Comment (Schmedu) [reply src/gluonts/mx/__init__.py]: Removed ✔️

File: src/gluonts/mx/__init__.py (hunk 2 above)
Comment (Schmedu) [reply src/gluonts/mx/__init__.py]: Sorted ✔️

File: docs/tutorials/forecasting/extended_tutorial.md (hunk 1 above)
Comment (Schmedu) [reply docs/tutorials/forecasting/extended_tutorial.md]: Concatenated ✔️

**Review by lostella — APPROVED** [review]
Looks good! Thanks!

File: src/gluonts/mx/__init__.py [hunk 4]
```diff
@@ -24,3 +24,11 @@
...
 
 from . import prelude as _  # noqa
+from .batchify import as_in_context, batchify  # noqa
+from .block.scaler import MeanScaler, NOPScaler  # noqa
+from .distribution import DistributionOutput, GaussianOutput  # noqa
+from .kernels import RBFKernel  # noqa
+from .model.estimator import GluonEstimator  # noqa
+from .model.predictor import RepresentableBlockPredictor  # noqa
+from .trainer import Trainer  # noqa
+from .util import copy_parameters, get_hybrid_forward_input_names  # noqa
```
Comment (jaheba) [inline src/gluonts/mx/__init__.py:34]: Something we could do is to define an `__all__`, which would allow us to remove the noqa-comments.

**Pull request merged**